### PR TITLE
[REQUIRES PREVIOUS PR COMPLETION] async pipe 

### PR DESCRIPTION
**Pull request by robbiemu**
THIS PR should not be completed until #10 is !

Swamy found a gem with the async methods.. especially in that it can generally be used with pipes and template methods.
Usually this works just like the Stateful service pattern that I came up with, but is more complex in the template and rxjs pipes that are required, but far less complex overall and less complex in the component.

Swamy please accept this code as a demo of what to review in your self study

### CODE REVIEW

**Comment by robbiemu** [comment]
@vishipayyallore having just removed the subscription property from each component, Im actually not positive that this is not still necessary in use cases where the component can be dynamically loaded. 

We should test to see if replacing the loaded component such as through cycling through 2 ng-templates will leave any trace of execution within the pipe when the component has been unloaded. I _believe_ that as long as we dont subscribe anywhere, we wont face that issue (the async pipe must handle the subscription for us and has onDestroy called when the component is unloaded)

**Review by vishipayyallore — APPROVED** [review]

**Comment by vishipayyallore** [comment]
@robbiemu 
Good morning. I have approved PR 10, and then approved PR 11. This PR is showing Conflicts. Could you please review it? Many thanks.

**Regards,
Swamy**

**Comment by robbiemu** [comment]
yup, thank Swamy. this is my fault, starting a branch from the previous pr branch. remerged main in and now resolved.

File: src/app/login/login.component.html [hunk 1]
```diff
@@ -1 +1,3 @@
-<button (click)="cta(copy.action)">{{copy.action}}</button>
\ No newline at end of file
+<ng-container *ngIf="action$|async as action">
```
Comment (robbiemu) [inline src/app/login/login.component.html:1]: @vishipayyallore please understand why this change

File: src/app/products/products.component.ts [hunk 2]
```diff
@@ -16,45 +16,23 @@ import { ProductsService } from './products.service'
   styleUrls: ['./products.component.scss']
 })
 @LogMethods({ when: !environment.production })
-export class ProductsComponent implements OnInit, OnDestroy {
-  subscriptions = new Subscription()
-  products: Array<Product> = []
-  hasProductsConfig?: boolean
+export class ProductsComponent implements OnInit {
+  products$!: Observable<Array<Product>>
```
Comment (robbiemu) [inline src/app/products/products.component.ts:20]: @vishipayyallore but you might start here, this is a simple, pristine example

**Review by vishipayyallore — COMMENTED** [review]
Approved
